core/src/saros/concurrent/management/JupiterClient.java

```diff
@@ -23,18 +23,15 @@ public JupiterClient(ISarosSession sarosSession) {
    * Jupiter instances for each local editor.
    *
    * @host and @client
+   *     <p>Note: This needs to be a *Concurrent*HashMap, because it is periodically iterated by the
+   *     HeartbeatDispatcher class.
    */
-  protected final HashMap<SPath, Jupiter> clientDocs = new HashMap<SPath, Jupiter>();
+  private final ConcurrentHashMap<SPath, Jupiter> clientDocs =
```

Comment:
```suggestion
  private final Map<SPath, Jupiter> clientDocs =
```

Reply:
I think @srossbach wanted this to be explicit to make sure, everybody gets, that this is *concurrent*.